src/contracts/core/DelegationManager.sol

```diff
@@ -785,6 +1018,146 @@ contract DelegationManager is
+    }
+
+    /**
+     * @notice helper pure to return scaledShares given shares and current totalMagnitude. Used for
+     * adding/removing staker shares from operatorScaledShares
+     */
+    function _scaleShares(uint256 shares, uint64 totalMagnitude) internal pure returns (uint256) {
+        return shares * SlashingConstants.PRECISION_FACTOR / totalMagnitude;
+    }
+
+    /**
+     * @notice helper pure to return real strategy shares given operator scaledShares and current totalMagnitude.
+     * Used for returning the total delegated shares for an operator and strategy
+     */
+    function _descaleShares(uint256 scaledShares, uint64 totalMagnitude) internal pure returns (uint256) {
+        return scaledShares * totalMagnitude / SlashingConstants.PRECISION_FACTOR;
+    }
+
+    /**
+     * @notice staker scaling factor should be initialized and lower bounded to 1e18
+     */
+    function _getStakerScalingFactor(address staker, IStrategy strategy) internal view returns (uint256) {
+        uint256 currStakerScalingFactor = stakerScalingFactors[staker][strategy];
+        if (currStakerScalingFactor == 0) {
+            currStakerScalingFactor = SlashingConstants.PRECISION_FACTOR;
+        }
+        return currStakerScalingFactor;
+    }
+
+    /**
+     * @notice helper to calculate the new staker scaling factor after adding shares. This is only used
+     * when a staker is depositing through the StrategyManager or EigenPodManager. A stakers scaling factor
+     * is only updated when they have new deposits and their shares are being increased.
+     */
+    function _calculateStakerScalingFactor(
+        address staker,
+        IStrategy strategy,
+        uint64 totalMagnitude,
+        uint256 existingShares,
+        uint256 addedShares
+    ) internal view returns (uint256) {
+        uint256 newStakerScalingFactor;
+
+        if (existingShares == 0) {
+            // existing shares are 0, meaning no existing delegated shares. In this case, the new staker scaling factor
+            // is re-initialized to
+            newStakerScalingFactor = SlashingConstants.PRECISION_FACTOR / (totalMagnitude);
+        } else {
+            uint256 currStakerScalingFactor = _getStakerScalingFactor(staker, strategy);
+
+            // TODO: DOUBLE CHECK THIS BEHAVIOR AND OVERFLOWS
+            // staker scaling factor is initialized to PRECISION_FACTOR(1e18) and totalMagnitude is initialized to INITIAL_TOTAL_MAGNITUDE(1e18)
+            // and is monotonically decreasing. You can deduce that the newStakerScalingFactor will never decrease to less than the PRECISION_FACTOR
+            // so this won't round to 0.
+            newStakerScalingFactor = (
+                currStakerScalingFactor * existingShares * totalMagnitude / SlashingConstants.PRECISION_FACTOR
+                    + addedShares * SlashingConstants.PRECISION_FACTOR
+            ) / ((existingShares + addedShares) * totalMagnitude);
```

Comment:
this should be
`(existingShares * currStakerScalingFactor / PRECISION_FACTOR * totalMagnitude / INITIAL_MAGNITUDE + addedShares) / ((existingShares + addedShares) * totalMagnitude / INITIAL_MAGNITUDE)`